### ISSUE TITLE
Future To do List

### ISSUE THREAD

**Issue by zorg2044** [zorg2044/BoringCrewServices#5]
- [ ] IVA

This will be after the initial 1.0 release

**Comment by EmberSkyMedia** [comment]
FreeIVA - check colliders (once IVA is complete).
Make sure you can get past the chute cover section, but not the lid if it’s closed.

**Comment by SofieBrink** [comment]
> Make sure you can get past the chute cover section, but not the lid if it’s closed.
I’m pretty sure that the chute cover won’t need any special attention as it isn’t anywhere in the passthrough chain since its attached to a node that’s separate from the docking port node on the capsule.
The same goes for the nose cap, and i also believe it to be impossible for an animation on one part to prevent passthrough on another. But will need @jonnyothan to verify that.

Either way the only scenario in which that’d matter would be if someone docked then subsequently closed the cap.

**Comment by JonnyOThan** [comment]
I'm not familiar with the configuration here, but yeah if you want the animation state of a part to affect traversibility then it would need to be in the attachNode chain.

**Comment by JacobB094** [comment]
Can we have the MFDs and keypads as separate display and control props, respectively? I think a lot could be done with that. Other props are easily configured using MAS/ASET, but there's currently no exact equivalent to the displays. I think MAS supports touchscreens as well. The keypad would need button colliders and, if you want to get fancy, animations, while the MFDs would need that and a screen surface.

**Comment by zorg2044** [comment]
> Can we have the MFDs and keypads as separate display and control props, respectively? I think a lot could be done with that. Other props are easily configured using MAS/ASET, but there's currently no exact equivalent to the displays. I think MAS supports touchscreens as well. The keypad would need button colliders and, if you want to get fancy, animations, while the MFDs would need that and a screen surface.

Yes I'm already getting some advice on how to do this. The MFDs, the keypads, the tablet, the abort handle and the rotational dials are all being considered as separate functional props. The joystick too if people want it.

I thought button animations were handled by the mod itself and is not a separate animation clip? But I havent looked at the details yet. I will take a look at all this once the basic IVA is in game as the functional props will be separate models using the same assets.

**Comment by JacobB094** [comment]
I think you can do it either way, but ASET uses animations defined within the model.

**Comment by JonnyOThan** [comment]
@zorg2044 The joystick should at least be a separate gameobject for VR support. You can make it a prop if you expect it to be useful in other locations too, or animated in response to player input (ASET has some examples)

**Comment by zorg2044** [comment]
> [@zorg2044](https://github.com/zorg2044) The joystick should at least be a separate gameobject for VR support. You can make it a prop if you expect it to be useful in other locations too, or animated in response to player input (ASET has some examples)

Alright I'll take a look, thanks

**Comment by EmberSkyMedia** [comment]
Suggested “interim” solution is use the Wyvern IVA with some minor adjustments to fit.

I may take a stab at it after I finish the revamp of stock biomes I’m working on.

**Comment by SofieBrink** [comment]
Feel free to do so and provide it as a seperate patch. I highly doubt we’d include it in any official BCS release as Zorg is hard at work on the official IVA and i don’t foresee another BCS release happening prior to that being finished

**Comment by zorg2044** [comment]
> Suggested “interim” solution is use the Wyvern IVA with some minor adjustments to fit.
> 
> I may take a stab at it after I finish the revamp of stock biomes I’m working on.

The new IVA is almost fully modelled and only texturing of the main cabin remains. Depending on my free time I expect it should be released in a month or two absolute worst case. I would rather not distribute any workaround in the meantime though you are free to make one for yourself of course.

![Image](https://github.com/user-attachments/assets/afa4a3ea-6b9c-4f9d-95fe-dd9130f0c8ef)

![Image](https://github.com/user-attachments/assets/08bae165-1ef6-4f2c-96ae-f5e7da6390c5)

**Comment by zorg2044** [comment]
Of course it will be available on dev for a few weeks for everyone to test before its officially released :) 

Honestly thought I would have it done this month but other stuff got in the way. Making steady progress though.

**Comment by EmberSkyMedia** [comment]
Just a heads up, as a temporary work around - not perfect and missing FreeIVA configs etc.
But its better than the black boxes that overlapped outside the capsule interferring with other IVAs.

Requires AcrAeroSpace 5 person Capsule installed

![Image](https://github.com/user-attachments/assets/20841d2b-bc43-4e9e-9f0c-6eb8f97f94b7)

`@PART[BCS_Centauri_CrewCapsule]
`{
`	!INTERNAL
`	@INTERNAL
`	{
`		name = ArcAero_Wyvern_IVA
`	}
`}
